sklearn/metrics/tests/test_classification.py

```diff
@@ -483,6 +483,34 @@ def test_matthews_corrcoef_multiclass():
     assert_almost_equal(mcc, 0.)
 
 
+def test_matthews_corrcoef_overflow():  # Bug#9622
+    def mcc_safe(y_true, y_pred):
+        conf_matrix = confusion_matrix(y_true, y_pred)
+        true_pos = conf_matrix[1,1]
+        false_pos = conf_matrix[1,0]
+        false_neg = conf_matrix[0,1]
+        n_points = float(conf_matrix.sum())
+        pos_rate = (true_pos + false_neg) / n_points
+        activity = (true_pos + false_pos) / n_points
+        mcc_numerator = true_pos / n_points - pos_rate * activity
+        mcc_denominator = activity * pos_rate * (1 - activity) * (1 - pos_rate)
+        return mcc_numerator / np.sqrt(mcc_denominator)
+
+    def random_ys(n_points):
+        x_true = np.random.sample(n_points)
+        x_pred = x_true + 0.2 * (np.random.sample(n_points) - 0.5)
+        y_true = (x_true > 0.5) * 1.0
+        y_pred = (x_pred > 0.5) * 1.0
+        return y_true, y_pred
+
+    for n_points in [100, 10000, 1000000]:
```

Comment:
Is there any reason to try with smaller numbers?

Reply:
The main reason is that the success at smaller numbers makes it clear that the failure with the big numbers indicates a problem with the stock function rather than with `mcc_safe`.
IOW, this is a "test of the test".
This is cheap, so I see no reason not to do it.